### DISCUSSION TITLE
get text copied to clipboard in headless mode does not work in chrome

### DISCUSSION THREAD

**Discussion by AlanShi01** [webdriverio/webdriverio#13840] [Community]
According to https://github.com/webdriverio/webdriverio/discussions/10810, it should be able to copy from clipboard in headless mode. But I tried the following code, and clipboardText is "". Can someone please help me with it? Thank you.
 
    await browser.url('http://localhost:8080');
    const puppeteerBrowser = await browser.getPuppeteer();
    // switch to Puppeteer
    const clipboardText = await browser.call(async () => {
        const context = puppeteerBrowser.browserContexts()[0];
        await context.overridePermissions(await browser.getUrl(), ['clipboard-read']);
        const pages = await puppeteerBrowser.pages();
        
        pages[0].click('button[aria-label="copy-to-clipboard"]');

        const clipboardText = (await pages[0].evaluate(() => navigator.clipboard.readText()));
        console.log("copied text: " + clipboardText);

        return clipboardText;
    })

    console.log(clipboardText) // returns ""

config is as follows

      capabilities: [
      {
     browserName: 'chrome',
     'goog:chromeOptions': {
      args:[
         '--headless=new',
         'disable-infobars',
         'disable-popup-blocking',
         'disable-notifications'
     ],
     prefs: {
         'profile.managed_default_content_settings.popups' : 2,
         'profile.managed_default_content_settings.notifications' : 2,
     }
     }
     }

**Comment by christian-bromann** [comment]
Hey @AlanShi01 , does this example work for you: https://github.com/webdriverio/example-recipes/blob/c2429ffd463a5666399e4e56a087484a6e2b22d5/keys/keys.js#L12 ?

**Reply by AlanShi01** [reply]
Hello @christian-bromann , thank you for your reply. Actually I need to test the "copy to clipboard" function.
